src/app/components/dropdown/dropdown.ts

```diff
@@ -1120,9 +1120,9 @@ export class Dropdown implements OnInit, AfterViewInit, AfterContentInit, AfterV
             const value = this.getOptionValue(option);
             this.updateModel(value, event);
             this.focusedOptionIndex.set(this.findSelectedOptionIndex());
-            isHide && setTimeout(() => this.hide(true), 1);
             preventChange === false && this.onChange.emit({ originalEvent: event, value: value });
         }
+        isHide && setTimeout(() => this.hide(true), 1);
```

Comment:
Hi @mehmetcetin01140 I think you can remove the `setTimeout` and the code will be cleaner.
Using this lines works perfectly:

```
if (isHide) {
   this.hide(true);
}
```

Here you have a repro:

![fixed](https://github.com/primefaces/primeng/assets/19764334/e3680cc2-7b89-404f-aef6-605076878ab8)